lips/lip-1.mdx

```diff
@@ -11,149 +11,89 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
 
-### Eventual Consistency
-The Off-Chain Protocol ensures eventual consistency of Shared Objects.
-
-# Basic Off-Chain Building Blocks
+**Reference ID**: Every Object type contains a `reference_id` field which is a unique reference ID for the Object. The reference ID is always specified by the payment initiator VASP (the VASP which originally created the Object). This value should be unique. We recommend the hexadecimal representation of a fresh >16 bytes random byte-string as a way of creating such values or deriving it as a UUID.
 
 ### On-Chain Account Setup
 A VASP's on-chain account must be created and set up in accordance with the standard VASP account creation process.
 
-Of particular note for off-chain APIs is the compliance key value.  This is the Ed25519 key with which a VASP signs travel rule statements on-chain and the required information within the KycDataObject exchanged via the Off-Chain Protocol.  During account creation, a VASP will set their compliance key which will be stored under the primary VASP account.
+Of particular note for off-chain APIs is the compliance key value.  This is the Ed25519 key with which a VASP signs travel rule statements on-chain and all off-chain requests and responses. During account creation, a VASP will set their compliance key which will be stored under the primary VASP account.
 
 ### HTTP end-point
 
-Each VASP exposes an HTTPS POST end point at `https://<hostname>:<port>/<protocol_version>/<LocalVASPAddress>/<RemoteVASPAddress>/command`.
+Each VASP exposes an HTTPS POST end point at `https://<hostname>:<port>/<protocol_version>/command`. The `protocol_version` is initially `v1` for this iteration of the off-chain APIs.
 
-* `protocol_version` is initially 0 for this iteration of the off-chain APIs.
-* `LocalVASPAddress` and `RemoteVASPAddress` are the bech32 VASP addresses of the respective VASPs - with the subaddress set to 0's as per [LIP 5](https://lip.libra.org/lip-5/).
-    * An example URL: http://127.0.0.1:7091/v1/lbr1pg9q5zs2pg9q5zs2pg9q5zs2pgyqqqqqqqqqqqqqqspa3m/lbr1pgfpyysjzgfpyysjzgfpyysjzggqqqqqqqqqqqqqxjt54z/command/
+### HTTP Headers
+All HTTP requests and responses must contain:
 
+* A header `X-REQUEST-ID` with a unique ID for the request, used for tracking requests and debugging. Responses must have the same string in the `X-REQUEST-ID` header value as the requests they correspond to.
+* A header `X-VERIFICATION-KEY-ADRRESS` with the LIP-5 Libra address of the VASP signing the HTTP reqquest or HTTP response. This header is used to retrieve the VASP verification key used to authenticate the request or response. For HTTP requests, the header is set to the VASP parent address initiating the request. For HTTP response, the header is set to the parent VASP address of the VASP responding.
```

Comment:
I changed my view with this `X-VERIFICATION-KEY-ADRRESS`. I think what should be here is just the acting VASP's address. Two reasons:

1. This helps with the receiving VASP to identity who it is talking to more quickly. It's possible that a VASP only wants to talk to VASP A, but not VASP B, even though they are all children of VASP C. Using `verification-key-address` makes VASPs need to dig into the payloads (and doing expensive crypto ops) to see the exact VASP address. This is not an imaginary scenario, Novi deals with this today.

2. A sound verification should check the address in the header matches the one in PaymentActor (which I think current implementation misses too?). Now this may be subjective, but checking `the one in header == the one in PaymentActor` is more intuitive than checking `the one in header parents the one in Payment Actor`.

The only advantage of having `verification-key-address` in header I think is the convenience of getting the compliance key. But stepping back a bit, wouldn't it be more convenient to directly attach the public key in header?

Reply:
use address will make sure it is a valid onchain vasp / dd account, given onchain account creation process. attach public key will make it open to everyone.

Reply:
My thoughts on this:

* I like the idea of having here the acting VASP, rather than the parent VASP. A correct implementation should check that the acting VASP has a payload signed by the parent VASP, and this indirection forces the check to retrieve the key.
* Due to the previous point, I would prefer to not attach the verification key. If we do this invited implementors to not check the binding between VASP - Parent VASP - verification key as they should.

PS I think the current offchain implementation does not deal with parent VASPs, and instead assumes it is given a direct tuple of (acting VASP, parent key). Then then checks do happen in the check-payment logic I believe.